### PR TITLE
Uses cudf 0.17

### PR DESCRIPTION
**Pull request by albert17**
New PR after #494 was closed and cannot be reopened (new_api branch does not exist anymore).

@rjzamora for info.

Errors found when using cudf 0.17:

1. `KeyError: None` in `tests/unit/test_dask_nvt.py::test_dask_workflow_api_dlrm`
    - This is due to a Dask bug. @rjzamora  has already submitted a [PR](https://github.com/dask/dask/pull/6969); and until we update to the next Dask version as a work around we can just avoid the flag `index=False` in NVTabular

2. `AssertionError: DataFrame.index` are different in `tests/unit/test_io.py::test_dask_dataset`
    - This error is happening when reading multiple parquet files. Trying to fix it adding the flag `check_index=False` since we don't care about matching indices in NVTabular. That solved the Parquet error, but creates errors for csv.

3. `assert <NA> is None` in `test_ops.py::test_difference_lag`
    - Having a frame with `dtype: bool`, we were doing the following operation `mask[mask == False] = None`. The resulting dataframe contains `<NA>` values instead of `None`. This is expected, cudf has stopped using None and instead is using `<NA>` to be coherent with pandas (`cudf.NA`).

4. `assert False` in `test_lamdaop`

5. `ValueError: Can only compare identically-labeled DataFrame` in `test_log` 

6. `RuntimeError: cuDF failure` in `test_normalize_minmax`, `test_fill_median`, and `test_normalize`
    - `RuntimeError: cuDF failure at: /opt/conda/envs/rapids/conda-bld/libcudf_1607613794356/work/cpp/src/binaryop/binaryop.cpp:695: Column sizes don't match`

 
**TODO: Fix errors**:
- [X] Fix `KeyError: None`
- [X] Fix `AssertionError: DataFrame.index`
- [X] Fix `assert <NA> is None`
- [X] Fix `assert False`
- [X] Fix `ValueError: Can only compare identically-labeled DataFrame`
- [X] Fix `RuntimeError: cuDF`

### CODE REVIEW

**Comment by albert17** [comment]
@benfred it works now for cudf 0.17.  

@rjzamora did most of the work, thank you very much!

File: tests/unit/test_ops.py [hunk 1]
```diff
@@ -640,16 +657,16 @@ def test_difference_lag():
     processor.fit(dataset)
     new_gdf = processor.transform(dataset).to_ddf().compute()
 
-    assert new_gdf["timestamp_difference_lag_1"][0] is None
+    assert new_gdf["timestamp_difference_lag_1"][0] is cudf.NA
```
Comment (rjzamora) [inline tests/unit/test_ops.py:660]: Perhaps you can add a `hasattr(cudf, "NA")` check for backward cudf compatibility?

Comment (albert17) [reply tests/unit/test_ops.py]: That makes sense. Added. Thanks Rick!

**Review by benfred — APPROVED** [review]
LGTM! Thanks @albert17 and @rjzamora !

**Pull request merged**